acpi/src/lib.rs

```diff
@@ -95,18 +117,19 @@ pub enum AcpiError {
     InvalidDsdtAddress,
     InvalidMadt(MadtError),
     InvalidGenericAddress,
+
+    AllocError,
 }
 
+/// Type capable of enumerating the existing ACPI tables on the system.
+///
+/// ### Remark: If using the `allocator_api` is an option, `PlatformInfo::new()` should be used
+///             to enumerate relevant ACPI structures, as it provides a much cleaner interface to
+///             the data.
 #[derive(Debug)]
-pub struct AcpiTables<H>
-where
-    H: AcpiHandler,
-{
-    /// The revision of ACPI that the system uses, as inferred from the revision of the RSDT/XSDT.
-    pub revision: u8,
-    pub sdts: BTreeMap<sdt::Signature, Sdt>,
-    pub dsdt: Option<AmlTable>,
-    pub ssdts: Vec<AmlTable>,
+pub struct AcpiTables<H: AcpiHandler> {
+    mapping: PhysicalMapping<H, SdtHeader>,
+    revision: u8,
```

Comment:
I think we should either keep this public, or add a getter for it?

Reply:
The mapping or the revision? If for the revision, I do agree, and I'll add one in.

Reply:
Yep just the revision